### PR TITLE
feat(sight): harden sqlite lifecycle

### PR DESCRIPTION
**Pull request by chengshuyi**
## Why

AgentSight recently hit a production incident where SQLite files consumed about 20 GB. The immediate growth bug is fixed, but lifecycle policy remained fragmented across stores, several databases had no bounded cleanup, and operators could not see the effective limits or current allocation.

## What changed

- add a business-model-free `agentsight-sqlite-lifecycle` crate for connection setup, DB/WAL/SHM measurement, checkpoint outcomes, retention cutoffs, lock retries, and adaptive size enforcement
- replace scattered storage settings with schema v3 `storage` policies and apply retention/capacity maintenance to primary, GenAI/evaluation, interruption, trajectory, optimization, and security-audit stores
- add authenticated `GET /api/storage/status` plus a Settings card showing effective policies, physical/logical usage, coverage, intervals, and maintenance owner without exposing filesystem paths
- update English/Chinese user guides, READMEs, architecture, and storage design documentation

## Related issue

no-issue: stability follow-up for the production SQLite disk-growth incident

## User / Agent impact

Operators can configure each AgentSight-owned SQLite store independently and inspect effective limits in Dashboard Settings. Config schema v3 intentionally replaces stale configuration with the shipped defaults after backing up the old file; old storage fields are not migrated.

## Risk and compatibility

- [x] Public CLI, API, configuration, or documented behavior changed
- [ ] Privileged or security-sensitive behavior changed
- [ ] Cross-component contract changed
- [x] Migration or rollback guidance is needed

The main compatibility change is `agentsight.json` schema v3 and its new `storage` section. Size enforcement deletes only store-owned eligible rows, never runs automatic `VACUUM`, and reports physical allocation separately from logical live-data usage. Rollback requires restoring the previous binary/config backup; existing SQLite table schemas remain readable.

## Validation

- `cargo fmt --all -- --check`
- `cargo clippy --all-targets -- -D warnings`
- `cargo test -p agentsight` — 1,830 passed, 1 ignored, plus binary/integration suites
- `cargo test -p agentsight-sqlite-lifecycle -p agentsight-opt-store -p agentsight-trajectory-collector -p agentsight-audit`
- `cargo test --workspace --exclude actplane-ifc-compiler`
- `cargo doc --workspace --no-deps`
- `npm run typecheck`, `npm run test:api-client`, `npm run test:i18n`, and `npm run build:embed` under `dashboard/`
- `bash scripts/docs-lint.sh`, `python3.11 scripts/docs-link-check.py`, and `npm run build --prefix website`
- launched `agentsight serve` against temporary storage; verified `/api/storage/status` returned all eight store classes without paths and the served Settings bundle contained the storage card
- L3 deep security review reported no findings

The unexcluded full workspace test command is independently blocked by the missing pre-existing `src/agentsight/test/policies` fixture used by `actplane-ifc-compiler`.

## Documentation and rollback

Updated both English and Chinese configuration, data/storage, and Dashboard guides, plus component READMEs and architecture documents. To roll back, restore the prior binary and the timestamped schema-v2 config backup; no database schema downgrade is required.

### CODE REVIEW

**Review by jfeng18 — COMMENTED** [review]
Deep review at `f0fef379` — the design and existing-fix preservation look solid; the incremental-coverage gate is the only blocker.

**Verified good (adversarial checks):**

- The #3378 migration-order fix in `interruption.rs` survives this refactor (ALTER still runs before the index batch, with the issue reference intact).
- Schema v3 upgrade is a shallow top-level-key merge with a timestamped backup; old storage fields become inert keys and the new `storage` section takes defaults — consistent with the PR description.
- `measure_database` deliberately measures without checkpointing, so "physical" means real on-disk allocation (main + WAL + SHM). That biases size enforcement toward earlier cleanup, which is the safe direction, and the intent is documented.
- `enforce_size_policy` is genuinely business-model-free (measure/delete/checkpoint injected as callbacks) and checkpointing after cleanup is part of the policy path.
- `recover_oom_events`, `maybe_purge_interruption_store`, and the genai `ensure_col` migrations all survive the unified.rs / schema.rs rewrites.

**The one blocker — incremental coverage < 80% over 2,033 new lines:**

- **`agentsight-audit/src/store/retention.rs` at 13.4% is the real gap**: `maintain()` (retention-day purge + size enforcement) is the core path this PR exists to protect, and it is nearly untested. Please add discriminating tests over retention boundary (`retention_days` = 0 / N) and size-trigger (below / above `limit_bytes`) behavior.
- `agentsight-trajectory-collector` (lib 54.3%, store 63.4%) and `agentsight-opt-store` (63.5%) maintenance paths should reach the threshold likewise.
- The server thin layers (`storage_status.rs` / `preferences.rs` / `mod.rs` route registrations at 0–8.9%) are low-value; either a minimal handler test or a documented maintainer waiver is fine.

With the retention/collector/opt-store maintenance paths covered, this is good to merge.

File: src/agentsight/src/storage/sqlite/genai/schema.rs [hunk 1]
```diff
@@ -222,256 +208,218 @@ impl GenAISqliteStore {
...
-                    break;
-                }
-                Ok(false) => {}
-                Err(e) => {
-                    log::warn!("WAL checkpoint failed on iteration {iterations}: {e}");
+        let report = enforce_size_policy::<Box<dyn std::error::Error>>(
+            policy,
+            || self.size_snapshot(),
+            |fraction| self.prune_old_records_with_percent(fraction),
+            || self.checkpoint_outcome(),
```
[the remark this answers is not in the document]
Comment (chengshuyi) [reply src/agentsight/src/storage/sqlite/genai/schema.rs:273]: Fixed in b04a20717. Every age-retention path now checkpoints before measuring/enforcing the size policy; a busy checkpoint stops further size pruning. Added a discriminating GenAI regression that holds a read snapshot and verifies an in-retention row is not deleted, plus an audit-store busy-checkpoint regression.

File: src/agentsight/Cargo.toml [hunk 2]
```diff
@@ -8,6 +8,7 @@ members = [
     "crates/agentsight-enforcer",
     "crates/agentsight-opt",
     "crates/agentsight-opt-store",
+    "crates/agentsight-sqlite-lifecycle",
```
[the remark this answers is not in the document]
Comment (chengshuyi) [reply src/agentsight/Cargo.toml:11]: Acknowledged. I kept this as one atomic incident follow-up because the shared lifecycle contract, per-store adoption, configuration, and observability must agree at each boundary. The maintainer deep review identified incremental coverage as the merge blocker; the exact CI coverage command now passes locally at 85%.

File: src/agentsight/src/config.rs [hunk 3]
```diff
@@ -376,10 +605,6 @@ pub struct JsonBatchConfig {
 #[serde(default)]
 pub struct JsonInterruptionFeature {
     pub enabled: Option<bool>,
```
[the remark this answers is not in the document]
Comment (chengshuyi) [reply src/agentsight/src/config.rs:609]: Fixed in b04a20717. Both interruption-detection guides now keep the feature toggle under features.interruption_detection and configure retention, size, and cadence under storage.interruptions.

**Comment by chengshuyi** [comment]
Addressed the deep-review blocker and refreshed this branch onto current `main`:\n\n- incremental Rust coverage now passes locally at **85%** (`2113` changed executable lines, `300` missing; threshold `80%`)\n- added direct maintenance coverage for security audit, trajectory, optimization, GenAI/evaluation, read-only store paths, and checkpoint-busy ordering\n- fixed age-retention ordering so a WAL checkpoint completes before size enforcement; a busy checkpoint now stops additional pruning\n- updated both interruption-detection guides to use `storage.interruptions`\n- resolved the latest `main` token CLI/storage conflicts while preserving its error-context behavior\n\nValidation: AgentSight tests, storage crate tests, fmt, clippy, architecture boundaries, Dashboard type/API/i18n/navigation tests, embedded build, docs lint, and link checks all pass locally.

**Review by jfeng18 — APPROVED** [review]
Re-reviewed at `a5f82b72` — the coverage blocker is resolved. Approving.

**Verified:**
- `maintain()` now has discriminating integration coverage: `maintenance_honors_disabled_and_enabled_retention` exercises the `retention_days` 0/N boundary (disabled keeps rows; enabled purges), and `maintenance_enforces_size_for_unreferenced_events` covers size enforcement.
- The new checkpoint ordering is the right call: after an age-retention purge, a busy checkpoint returns `CheckpointBusy` and stops further pruning instead of charging ahead with stale measurements.
- trajectory-collector (26 tests in store.rs) and opt-store (10 in lib.rs) maintenance paths are now covered; the #3378 migration-order fix survives the rebase.
- CI fully green (36/36), which also confirms the incremental-coverage gate passes.

Thanks for the quick turnaround — this closes out the SQLite lifecycle hardening nicely.
